### PR TITLE
Default predeploy backup option to production envs only

### PR DESCRIPTION
**Pull request by halkyon**

### CODE REVIEW

File: code/backends/CapistranoDeploymentBackend.php [hunk 1]
```diff
@@ -154,9 +154,9 @@ public function deploy(
 	/**
 	 * @return array
 	 */
-	public function getDeployOptions() {
+	public function getDeployOptions(DNEnvironment $environment) {
 		return [
-			new PredeployBackupOption(true),
+			new PredeployBackupOption($environment->Usage === 'Production'),
```
Comment (stojg) [inline code/backends/CapistranoDeploymentBackend.php]: We should change the magic string to a constant for `Usage`, before we start mispeling `prod`, `Production` or `production`

File: code/backends/DemoDeploymentBackend.php [hunk 2]
```diff
@@ -67,7 +67,7 @@ public function deploy(
 	/**
 	 * @return array
 	 */
-	public function getDeployOptions() {
+	public function getDeployOptions(DNEnvironment $environment) {
```
Comment (stojg) [inline code/backends/DemoDeploymentBackend.php]: missing docblock for argument

File: code/backends/CapistranoDeploymentBackend.php (hunk 1 above)
Comment (halkyon) [inline code/backends/CapistranoDeploymentBackend.php]: Done!

File: code/backends/DemoDeploymentBackend.php (hunk 2 above)
Comment (halkyon) [inline code/backends/DemoDeploymentBackend.php]: Done!

**Pull request merged**